screensharing/main.js

```diff
@@ -0,0 +1,113 @@
+        this._onScreenSharingEvent = this._onScreenSharingEvent.bind(this);
+
+        // Listen for events coming in from the main render window and the screen share tracker window.
+        ipcMain.on(SCREEN_SHARE_EVENTS_CHANNEL, this._onScreenSharingEvent);
+
+        // Clean up ipcMain handlers to avoid leaks.
+        app.once('window-all-closed', () => {
+            ipcMain.removeListener(SCREEN_SHARE_EVENTS_CHANNEL, this._onScreenSharingEvent);
+        });
+    }
+
+    /**
+     * Listen for events coming on the screen sharing event channel.
+     *
+     * @param {Object} event - Electron event data.
+     * @param {Object} data - Channel specific data.
+     */
+    _onScreenSharingEvent(event, { data }) {
+        switch (data.name) {
+            case SCREEN_SHARE_EVENTS.OPEN_TRACKER:
+                this._createScreenShareTracker();
+                break;
+            case SCREEN_SHARE_EVENTS.CLOSE_TRACKER:
+                if (this._screenShareTracker) {
+                    this._screenShareTracker.close();
+                    this._screenShareTracker = undefined;
+                }
+                break;
+            case SCREEN_SHARE_EVENTS.STOP_SCREEN_SHARE:
+                this._jitsiMeetWindow.webContents.send(SCREEN_SHARE_EVENTS_CHANNEL, { data });
+                break;
+            default:
+                console.warn(`Unhandled ${SCREEN_SHARE_EVENTS_CHANNEL}: ${data}`);
+        }
+    }
+
+    /**
+     * Opens an always on top window, in the bottom center of the screen, that lets a user know
+     * a content sharing session is currently active.
+     *
+     * @return {void}
+     */
+    _createScreenShareTracker() {
+        if (this._screenShareTracker) {
+            return;
+        }
+
+        // Display always on top screen sharing tracker window in the center bottom of the screen.
+        let display = screen.getPrimaryDisplay();
+        this._screenShareTracker = new BrowserWindow({
+            height: TRACKER_SIZE.height,
+            width: TRACKER_SIZE.width,
+            x:(display.bounds.width - TRACKER_SIZE.width) / 2,
+            y:display.bounds.height - TRACKER_SIZE.height - 10,
+            transparent: true,
+            minimizable: true,
+            maximizable: false,
+            resizable: false,
+            alwaysOnTop: true,
+            fullscreen: false,
+            fullscreenable: false,
+            skipTaskbar: true,
+            frame: false,
+            show: false,
+            webPreferences: {
+                nodeIntegration: true
+            }
+        });
+
+        // Prevent newly created window to take focus from main application.
+        this._screenShareTracker.once('ready-to-show', () => {
+            if (this._screenShareTracker && !this._screenShareTracker.isDestroyed()) {
+                this._screenShareTracker.showInactive();
+            }
+        });
+
+        this._screenShareTracker.sharingIdentity = this._identity;
```

Comment:
Well, looked around and it seems to be an ok way to pass parameters.

Reply:
But is it documented somewhere? 
Do you know if those custom props are not lost internally at some point?
Does it work on all electron versions?

NOTE: IMHO if it is not documented I'm worried that this may stop working at some point or that it is not always working even now. Also on the top of my head you can very easily pass it trough the IPC objects or define it as a global even. Probably there are even better ways to pass it that match the official electron APIs.

Reply:
changed to o more accepted approach.